ownership-chain/pallets/parachain-staking/src/traits.rs

```diff
@@ -47,23 +47,23 @@ impl OnNewRound for () {
 }
 
 /// Defines the behavior to payout the collator's reward.
-pub trait PayoutCollatorReward<Runtime: crate::Config> {
-	fn payout_collator_reward(
+pub trait PayoutReward<Runtime: crate::Config> {
+	fn payout_reward(
 		round_index: crate::RoundIndex,
 		collator_id: Runtime::AccountId,
 		amount: crate::BalanceOf<Runtime>,
-	) -> Weight;
+	) -> Result<crate::BalanceOf<Runtime>, DispatchError>;
```

Comment:
this should still return the weight it consumed. this function is used in `on_initialize`, so it's very important to keep track of the weights. I propose returning tuple `(Balance, Weight)` and handle the error in the function. or return `Result<(Balance, Weight), Error>`

Reply:
now weight is recorded here: https://github.com/freeverseio/laos/blob/20af53a43dee40cd0de5daeb9e7f473af44ccfa8/ownership-chain/pallets/parachain-staking/src/lib.rs
you can see in the previous implementation this function is called but zero weight is returned: https://github.com/freeverseio/laos/blob/main/ownership-chain/pallets/parachain-staking/src/traits.rs#L36

Reply:
isn't that default impl? did they use default implementation?

Reply:
> now weight is recorded here:
maybe unify the logic with payout reward and return the weight in a tuple with amount?

Reply:
solved by https://github.com/freeverseio/laos/pull/411